Linux 6.15.3上でDKMSコンパイルが失敗する

objtoolのIBT関連の[パッチ](https://lkml.org/lkml/2025/3/24/1428)により、古い名前を使っているinit_module()やcleanup_module() (多分[これ？](https://github.com/tsukumijima/px4_drv/blob/12a1213912a100f8bd14fccf92180ad3955f6037/driver/driver_module.c#L18))
が存在すると、DKMSがエラーを吐くことがわかりました。

以下がエラーログ全文になります
```
[reimu@gensokyo px4_drv]$ sudo dkms install px4_drv/0.5.4
Deprecated feature: CLEAN (/var/lib/dkms/px4_drv/0.5.4/source/dkms.conf)
Sign command: /usr/lib/modules/6.15.3-arch1-1/build/scripts/sign-file
Signing key: /var/lib/dkms/mok.key
Public certificate (MOK): /var/lib/dkms/mok.pub

Building module(s).....(bad exit status: 2)
Failed command:
cd ./driver; make KVER=6.15.3-arch1-1 px4_drv.ko

Error! Bad return status for module build on kernel: 6.15.3-arch1-1 (x86_64)
Consult /var/lib/dkms/px4_drv/0.5.4/build/make.log for more information.
[reimu@gensokyo px4_drv]$ cat /var/lib/dkms/px4_drv/0.5.4/build/make.log 
DKMS (dkms-3.2.1) make.log for px4_drv/0.5.4 for kernel 6.15.3-arch1-1 (x86_64)
Tue Jun 24 11:42:01 PM JST 2025

Building module(s)
# command: cd ./driver; make KVER=6.15.3-arch1-1 px4_drv.ko
'revision.h' was updated.
make[1]: Entering directory '/usr/lib/modules/6.15.3-arch1-1/build'
make[2]: Entering directory '/var/lib/dkms/px4_drv/0.5.4/build/driver'
  CC [M]  driver_module.o
  CC [M]  ptx_chrdev.o
  CC [M]  px4_usb.o
  CC [M]  px4_usb_params.o
  CC [M]  px4_device.o
  CC [M]  px4_device_params.o
  CC [M]  px4_mldev.o
  CC [M]  pxmlt_device.o
  CC [M]  isdb2056_device.o
  CC [M]  it930x.o
  CC [M]  itedtv_bus.o
  CC [M]  tc90522.o
  CC [M]  r850.o
  CC [M]  rt710.o
  CC [M]  cxd2856er.o
  CC [M]  cxd2858er.o
  CC [M]  ringbuffer.o
  CC [M]  s1ur_device.o
  CC [M]  m1ur_device.o
  LD [M]  px4_drv.o
px4_drv.o: error: objtool: cleanup_module(): Magic init_module() function name is deprecated, use module_init(fn) instead
make[4]: *** [/usr/lib/modules/6.15.3-arch1-1/build/scripts/Makefile.build:408: px4_drv.o] Error 255
make[4]: *** Deleting file 'px4_drv.o'
make[3]: *** [/usr/lib/modules/6.15.3-arch1-1/build/Makefile:2003: .] Error 2
make[2]: *** [/usr/lib/modules/6.15.3-arch1-1/build/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/var/lib/dkms/px4_drv/0.5.4/build/driver'
make[1]: *** [Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/usr/lib/modules/6.15.3-arch1-1/build'
make: *** [Makefile:19: px4_drv.ko] Error 2

# exit code: 2
# elapsed time: 00:00:09
----------------------------------------------------------------
[reimu@gensokyo px4_drv]$ uname -a
Linux gensokyo 6.15.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 19 Jun 2025 14:41:19 +0000 x86_64 GNU/Linux
```

回避策として、新しい名前に書き換えれば動く・・・と思うので、(DKMS関連はちょっとわからないので)、確認お願いいたします

--- a/driver/driver_module.c
+++ b/driver/driver_module.c
@@ -10,12 +10,17 @@
 
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/version.h>
 
 #include "revision.h"
 #include "px4_usb.h"
 #include "firmware.h"
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,4)
 static int m_init(void)
+#else
+int init_module(void)
+#endif
 {
 	int ret = 0;
 
@@ -47,14 +52,19 @@ static int m_init(void)
 	return 0;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,4)
 static void m_cleanup(void)
+#else
+void cleanup_module(void)
+#endif
 {
 	px4_usb_unregister();
 }
 
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,4)
 module_init(m_init);
 module_exit(m_cleanup);
+#endif
 
 MODULE_VERSION(PX4_DRV_VERSION);
 MODULE_AUTHOR("nns779");

--- a/driver/driver_module.c
+++ b/driver/driver_module.c
@@ -15,7 +15,7 @@
 #include "px4_usb.h"
 #include "firmware.h"
 
-int init_module(void)
+static int m_init(void)
 {
 	int ret = 0;
 
@@ -47,10 +47,14 @@ int init_module(void)
 	return 0;
 }
 
-void cleanup_module(void)
+static void m_cleanup(void)
 {
 	px4_usb_unregister();
 }
+
+
+module_init(m_init);
+module_exit(m_cleanup);
 
 MODULE_VERSION(PX4_DRV_VERSION);
 MODULE_AUTHOR("nns779");

